secretservice/secretservice.go

```diff
@@ -0,0 +1,305 @@
+package secretservice
+
+import (
+	"fmt"
+	"math/big"
+	"time"
+
+	dbus "github.com/guelfey/go.dbus"
+	errors "github.com/pkg/errors"
+)
+
+const SecretServiceInterface = "org.freedesktop.secrets"
+const SecretServiceObjectPath dbus.ObjectPath = "/org/freedesktop/secrets"
+const DefaultCollection dbus.ObjectPath = "/org/freedesktop/secrets/aliases/default"
+
+type authenticationMode string
+
+const AuthenticationPlain authenticationMode = "plain"
+const AuthenticationDHIETF1024SHA256AES128CBCPKCS7 authenticationMode = "dh-ietf1024-sha256-aes128-cbc-pkcs7"
+
+const NilFlags = 0
+
+type Attributes map[string]string
+
+type Secret struct {
+	Session     dbus.ObjectPath
+	Parameters  []byte
+	Value       []byte
+	ContentType string
+}
+
+type PromptCompletedResult struct {
+	Dismissed bool
+	Paths     dbus.Variant
+}
+
+type SecretService struct {
+	conn     *dbus.Conn
+	signalCh <-chan *dbus.Signal
+}
+
+type Session struct {
+	Mode    authenticationMode
+	Path    dbus.ObjectPath
+	Public  *big.Int
+	Private *big.Int
+	AESKey  []byte
+}
+
+func NewService() (*SecretService, error) {
```

Comment:
It would be great to have tests for this too.  I assume they will only work on linux, so add

    // +build linux

to the test file.

Reply:
I can add tests but I don't think they'll work in CI since the system needs to have a keyring running, is that ok?

Reply:
ok